stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/FileAttachmentFactory.kt

```diff
@@ -0,0 +1,127 @@
+package io.getstream.chat.android.compose.ui.attachments
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.CloudDownload
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import com.getstream.sdk.chat.utils.MediaStringUtil
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.compose.R
+import io.getstream.chat.android.compose.state.messages.attachments.AttachmentState
+import io.getstream.chat.android.compose.ui.attachments.components.FileAttachmentImage
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.offline.ChatDomain
+
+/**
+ * An extension of the [AttachmentFactory] that validates attachments as files and uses [FileAttachmentContent] to
+ * build the UI for the message.
+ * */
+public class FileAttachmentFactory : AttachmentFactory() {
+
+    override fun predicate(message: Message): Boolean {
+        return message.attachments.any { it.type != "image" }
+    }
+
+    @Composable
+    public override fun Content(attachmentState: AttachmentState) {
+        FileAttachmentContent(attachmentState)
+    }
+}
+
+/**
+ * Builds a file attachment message.
+ *
+ * @param attachmentState - The state of the attachment, holding the root modifier, the message
+ * and the onLongItemClick handler.
+ * */
+@Composable
+public fun FileAttachmentContent(attachmentState: AttachmentState) {
+    val (modifier, messageItem, _) = attachmentState
+    val (message, _) = messageItem
+
+    Column(
+        modifier = modifier
+            .wrapContentHeight()
+            .width(200.dp)
```

Comment:
im guessing since this is internal `message` UI so we don't allow customization , would this be useful  for later usecase

Reply:
Yeah this is an internal thing - the default UI/behavior we expose for attachments. We can still customize it through some ChatTheme params, like the `attachmentShape` and through the modifier.

We could expose more customization, but if people want completely custom items, they can just override the factory overall.